dataset/shape.cpp

```diff
@@ -254,4 +254,14 @@ DataArray flatten(const DataArray &a, const scipp::span<const Dim> &from_labels,
   });
 }
 
+DataArray transpose(const DataArray &a, const std::vector<Dim> &dims) {
+  return {transpose(a.data(), dims), a.coords(), a.masks(), a.attrs(),
```

Comment:
So what happens if you have a 2D coord, should it also be transposed?
And a 2D mask?
I know dimensions can be flipped between data and masks, but it would be counter-intuitive to users i think to only see the data get transposed?
In the case of 2D coords, there is the extra implication as to which dim (the inner one?) is the underlying dimension.

Reply:
Parts of the answer is the discussion in the linked issue.

We cannot transpose coords in general since that may not work for non-dimension binned coords.

> In the case of 2D coords, there is the extra implication as to which dim (the inner one?) is the underlying dimension.
Exactly, but it is the inner dim *of the coord*, not the inner dim *of the data*, i.e., we have to keep the dim order for coords.

For masks, at this point I see no need for the extra complication of transposing the masks. This would only be relevant if someone then works directly with the masks.